src/index.js

```diff
@@ -1,7 +1,349 @@
+      currentLng: start[0],
+      initialLat: start[1],
+      initialLng: start[0],
+      moveArr: this.state.moveArr.concat([[start[1], start[0]]]),
+      zoom: 18,
+    });
+
+    //Gives a random lat and lng between the max and min lat and lng of Vermont.
+    function randomLocation() {
+      let randomLng =
+        -1 * (Math.random() * (71.510225 - 73.352182) + 73.352182);
+      let randomLat = Math.random() * (45.005419 - 42.730315) + 42.730315;
+
+      let stateBorder = L.geoJson(borderData);
+
+      let results = leafletPip.pointInLayer(
+        [randomLng, randomLat],
+        stateBorder
+      );
+
+      while (results.length === 0) {
+        randomLng = -1 * (Math.random() * (71.510225 - 73.352182) + 73.352182);
+        randomLat = Math.random() * (45.005419 - 42.730315) + 42.730315;
+        results = leafletPip.pointInLayer([randomLng, randomLat], stateBorder);
+      }
+
+      return [randomLng, randomLat];
+    }
+  };
+
+//Opens the modal
+  handleGuess=(evt)=>{
+    evt.preventDefault();
+    this.setState({
+      modal: true,
+
+    })
+  }
+
+//This function closes the modal
+  closeModal = (evt) => {
+    evt.preventDefault();
+    this.setState(() => {
+      return { modal: false,
+      winModal: false};
+    })};
+
+//Creating a scores object and updating the array of all scores.
+  submitName = (evt) => {
+    evt.preventDefault()
+    let scores = JSON.parse(localStore.getItem("scores")) || []
+    let stringScore = this.state.score.toString()
+    let scoreObj = {
+      name: this.state.userName,
+      score: stringScore,
+    }
+    scores.push(scoreObj)
+    localStore.setItem("scores", JSON.stringify(scores))
+    this.setState({
+      winModal: false
+    })
+    
+  }
+
+  //Sets the username
+  handleChange = (evt) => {
+    evt.preventDefault()
+    let userName = evt.target.value;
+    this.setState({
+      userName: userName,
+      
+    })
+  }
+
+
+  //This function allows user to make a guess. Fetches the county, if the guess is the same county displays score. If guess is incorrect, subtracts ten from score and displays score
+  makeGuess = (evt) => {
+    evt.preventDefault()
+
+    let selectedCounty = document.getElementById("countyList").value
+    let currentScore = this.state.score;
+    
+
+    fetch(`https://nominatim.openstreetmap.org/reverse?lat=${this.state.initialLat}&lon=${this.state.initialLng}&format=geojson`)
+    .then((res) => res.json())
+    .then((json) => {
+      console.log(json.features[0].properties.address.county)
+      if (selectedCounty === json.features[0].properties.address.county) {
+        this.setState({
+          startDisabled: false,
+          guessDisabled: true,
+          quitDisabled: true,
+          town:
+            (json.features[0].properties.address.city ||
+            json.features[0].properties.address.town ||
+            json.features[0].properties.address.village ||
+            json.features[0].properties.address.hamlet),
```

Comment:
You would want these to be in reverse order. The code will display the first one that it hits and every one of these places will have a city property. 

As it is right now your code will never display the town, village, or hamlet properties

Reply:
They don't all have a city property, which is why we included the town, village and hamlet. As it turns out, hamlet is just a sub-section of a town/village so it's unnecessary, but the reason we had city first is because that was the most common municipality property name so we figured we might as well put that first since it is going to be hit most often. But it is definitely hitting town and village when necessary because city/village/town are mutually exclusive. I'll remove hamlet though, as that one isn't getting hit.